geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java

```diff
@@ -92,13 +93,328 @@ public static void tearDown() {
+
+    jedis.set(key, value, setParams);
+
+    Long result = jedis.ttl(key);
+
+    assertThat(result).isGreaterThan(15l);
+  }
+
+  @Test
+  public void testSET_with_Negative_PX_time_shouldReturnError() {
+    String key = "key";
+    String value = "value";
+    int millisecondsUntilExpiration = -1;
+
+    SetParams setParams = new SetParams();
+    setParams.px(millisecondsUntilExpiration);
+
+    assertThatThrownBy(() -> jedis.set(key, value, setParams))
+        .isInstanceOf(JedisDataException.class)
+        .hasMessageContaining(RedisConstants.ERROR_INVALID_EXPIRE_TIME);
+  }
+
+  @Test
+  public void testSET_shouldClearPreviousTTL_onSuccess() {
+    String key = "key";
+    String value = "value";
+    int secondsUntilExpiration = 20;
+
+    SetParams setParams = new SetParams();
+    setParams.ex(secondsUntilExpiration);
+
+    jedis.set(key, value, setParams);
+
+    jedis.set(key, "other value");
+
+    Long result = jedis.ttl(key);
+
+    assertThat(result).isEqualTo(-1L);
+  }
+
+  @Test
+  public void testSET_shouldClearPreviousTTL_onXXSuccess() {
+    String key = "xx_key";
+    String value = "did exist";
+    int secondsUntilExpiration = 20;
+    SetParams setParamsXX = new SetParams();
+    setParamsXX.xx();
+    SetParams setParamsEX = new SetParams();
+    setParamsEX.ex(secondsUntilExpiration);
+    String result_EX = jedis.set(key, value, setParamsEX);
+    assertThat(result_EX).isEqualTo("OK");
+    assertThat(jedis.ttl(key)).isGreaterThan(15L);
+
+    String result_XX = jedis.set(key, value, setParamsXX);
+
+    assertThat(result_XX).isEqualTo("OK");
+    Long result = jedis.ttl(key);
+    assertThat(result).isEqualTo(-1L);
+  }
+
+  @Test
+  public void testSET_should_not_clearPreviousTTL_onFailure() {
+    String key_NX = "nx_key";
+    String value_NX = "set only if key did not exist";
+    int secondsUntilExpiration = 20;
+
+    SetParams setParamsEX = new SetParams();
+    setParamsEX.ex(secondsUntilExpiration);
+
+    SetParams setParamsNX = new SetParams();
+    setParamsNX.nx();
+
+    jedis.set(key_NX, value_NX, setParamsEX);
+    String result_NX = jedis.set(key_NX, value_NX, setParamsNX);
+    assertThat(result_NX).isNull();
+
+    Long result = jedis.ttl(key_NX);
+    assertThat(result).isGreaterThan(15L);
+  }
+
+  @Test
+  @Ignore
+  public void testSET_with_KEEPTTL_shouldRetainPreviousTTL_OnSuccess() {
+    String key = "key";
+    String value = "value";
+    int secondsToExpire = 30;
+
+    SetParams setParamsEx = new SetParams();
+    setParamsEx.ex(secondsToExpire);
+
+    jedis.set(key, value, setParamsEx);
+
+    SetParams setParamsKeepTTL = new SetParams();
+    // setParamsKeepTTL.keepTtl();
+    // Jedis Doesn't support KEEPTTL yet.
+
+    jedis.set(key, "newValue", setParamsKeepTTL);
+
+    Long result = jedis.ttl(key);
+    assertThat(result).isGreaterThan(15L);
+  }
+
+  @Test
+  public void testSET_shouldAccept_NX_argumentToOnlySetKeyIfIt_Does_not_Exist() {
```

Comment:
the names of these tests are inconsistent:
here Does_not_Exist has underscores when the convention suggests no underscores and camelCase
the word HONOR used above seems inconsistent with the other language of the the test names.

Reply:
done